openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ServiceToOpenAPIConverterUtils.java

```diff
@@ -191,44 +205,27 @@ private static void extractListenersAndServiceNodes(String serviceName, List<Str
-     * The new API provides a list of {@code OASResult}, which contains all the diagnostic information collected
-     * while generating the openAPI contract.
-     */
-    @Deprecated
-    public static String generateOASForGivenFormat(ServiceDeclarationNode serviceDeclarationNode,
-                                                   boolean needJson, List<ListenerDeclarationNode> endpoints,
-                                                   SemanticModel semanticModel, String openApiName) {
-        Optional<OpenAPI> openapi = generateOAS(serviceDeclarationNode, endpoints, semanticModel, openApiName,
-                null).getOpenAPI();
-        if (openapi.isPresent()) {
-            OpenAPI openAPI = openapi.get();
-            if (openAPI.getInfo().getTitle() == null || openAPI.getInfo().getTitle().equals(SLASH)) {
-                openAPI.getInfo().setTitle(normalizeTitle(openApiName));
-            }
-            return needJson ? Json.pretty(openAPI) : Yaml.pretty(openAPI);
-        }
-        return "Error while generating openAPI contract";
-    }
-
     /**
      * Provides an instance of {@code OASResult}, which contains the generated contract as well as
      * all the diagnostics information.
      *
-     * @param serviceDefinition     Service Node related to ballerina service
-     * @param endpoints             Listener endpoints that bind to service
-     * @param semanticModel         Semantic model for given ballerina file
-     * @param openApiFileName       OpenAPI file name
-     * @param ballerinaFilePath     Input ballerina file Path
+     * @param oasGenerationMetaInfo    Includes the service definition node, endpoints, semantic model, openapi file
+     *                                 name and ballerina file path
      * @return {@code OASResult}
      */
-    public static OASResult generateOAS(ServiceDeclarationNode serviceDefinition,
-                                        List<ListenerDeclarationNode> endpoints, SemanticModel semanticModel,
-                                        String openApiFileName, Path ballerinaFilePath) {
+    public static OASResult generateOAS(OASGenerationMetaInfo oasGenerationMetaInfo) {
         // 01.Fill the openAPI info section
+        return ensembleOASResult(oasGenerationMetaInfo.getServiceDeclarationNode(),
+                oasGenerationMetaInfo.getEndpoints(),
+                oasGenerationMetaInfo.getSemanticModel(), oasGenerationMetaInfo.getOpenApiFileName(),
+                oasGenerationMetaInfo.getBallerinaFilePath());
+    }
+
+    private static OASResult ensembleOASResult(ServiceDeclarationNode serviceDefinition,
+                                               LinkedHashSet<ListenerDeclarationNode> endpoints,
+                                               SemanticModel semanticModel,
+                                               String openApiFileName, Path ballerinaFilePath) {
+
```

Comment:
Do we need 2 methods for this? Can't we just extract the `oasGenerationMetaInfo` within the same function and use it?

Reply:
Yes, we can implement it within one function.